server/services/store/storetests/blocks.go

```
@@ -310,3 +317,128 @@ func testGetRootID(t *testing.T, store store.Store, tearDown func()) {
+	require.NoError(t, err)
+	require.Len(t, blocks, initialCount+6)
+
+	t.Run("from root id", func(t *testing.T) {
+		parentID, err := store.GetParentID("parent")
+		require.NoError(t, err)
+		require.Equal(t, "", parentID)
+	})
+
+	t.Run("from child id", func(t *testing.T) {
+		parentID, err := store.GetParentID("grandchild1")
+		require.NoError(t, err)
+		require.Equal(t, "child1", parentID)
+	})
+
+	t.Run("from not existing id", func(t *testing.T) {
+		_, err := store.GetParentID("not-exists")
+		require.Error(t, err)
+	})
+}
+
+func testDeleteBlock(t *testing.T, store store.Store) {
+	userID := "user-id"
+
+	blocks, err := store.GetAllBlocks()
+	require.NoError(t, err)
+	initialCount := len(blocks)
+
+	blocksToInsert := []model.Block{
+		{
+			ID:         "block1",
+			RootID:     "block1",
+			ModifiedBy: userID,
+		},
+		{
+			ID:         "block2",
+			RootID:     "block2",
+			ModifiedBy: userID,
+		},
+		{
+			ID:         "block3",
+			RootID:     "block3",
+			ModifiedBy: userID,
+		},
+	}
+	InsertBlocks(t, store, blocksToInsert)
+
+	blocks, err = store.GetAllBlocks()
+	require.NoError(t, err)
+	require.Len(t, blocks, initialCount+3)
+
+	t.Run("exiting id", func(t *testing.T) {
+		// Wait for not colliding the ID+insert_at key
+		time.Sleep(1 * time.Millisecond)
+		err := store.DeleteBlock("block1", userID)
+		require.NoError(t, err)
+	})
+
+	t.Run("exiting id multiple times", func(t *testing.T) {
+		// Wait for not colliding the ID+insert_at key
+		time.Sleep(1 * time.Millisecond)
+		err := store.DeleteBlock("block1", userID)
+		require.NoError(t, err)
+		// Wait for not colliding the ID+insert_at key
+		time.Sleep(1 * time.Millisecond)
+		err = store.DeleteBlock("block1", userID)
+		require.NoError(t, err)
+	})
+
+	t.Run("from not existing id", func(t *testing.T) {
+		// Wait for not colliding the ID+insert_at key
+		time.Sleep(1 * time.Millisecond)
+		err := store.DeleteBlock("not-exists", userID)
+		require.NoError(t, err)
```

@chenilim Is this the expected behavior?

---

Yes, same as above. Deleting something that doesn't exist won't impact the model, except to bloat the database. It's a "silent failure" on the client. Have to think more about unintended effects, but I think this is the right behavior for now.